### PR TITLE
docker-completion: build from github source tarball

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

```
$ /opt/homebrew/Cellar/docker/26.1.3/bin/docker --version
Docker version 26.1.3, build Homebrew
```

followup #171906

### CODE REVIEW

File: Formula/d/docker.rb [hunk 1]
```diff
@@ -35,7 +34,7 @@ def install
     ldflags = %W[
       -s -w
       -X github.com/docker/cli/cli/version.BuildTime=#{time.iso8601}
-      -X github.com/docker/cli/cli/version.GitCommit=#{Utils.git_short_head}
+      -X github.com/docker/cli/cli/version.GitCommit=#{tap.user}
```
Comment (carlocab) [inline Formula/d/docker.rb:37]: We can probably keep this in case users rely on it. But we can definitely install the completions from a tarball.

Comment (chenrui333) [reply Formula/d/docker.rb:37]: why does people reply on it? what is the use case?

Comment (carlocab) [reply Formula/d/docker.rb:37]: Debugging, perhaps? Presumably if there were no use case upstream wouldn't build this into the output of their version information...

Comment (chenrui333) [reply Formula/d/docker.rb:37]: for release build, it is not useful (the [vcs provider introduced in go1.18](https://tip.golang.org/doc/go1.18#debug/buildinfo) for debugging build), that is why I am removing it for all golang builds.

Comment (SMillerDev) [reply Formula/d/docker.rb:37]: I've seen quite some arguments from upstream developers that disagree with that. They generally want to know what commit the versioned build was based on.

Comment (MikeMcQuaid) [reply Formula/d/docker.rb:37]: We should definitely keep this. I see no reason to ever override `GitCommit` to not mention a Git commit and, if this has been done on any other formulae, these changes should be reverted.

```suggestion
      -X github.com/docker/cli/cli/version.GitCommit=#{Utils.git_short_head}-Homebrew
```

or something if you really want to identify Homebrew usage here.

Comment (chenrui333) [reply Formula/d/docker.rb:37]: > I've seen quite some arguments from upstream developers that disagree with that. They generally want to know what commit the versioned build was based on.

The version is equivalent to the commit, if we are going to do it, should we revert all the go formulae to the git checkout. (maybe another RFC on this topic?)

Comment (MikeMcQuaid) [reply Formula/d/docker.rb]: I don't think it is equivalent.



> should we revert all the go formulae to the git checkout.

or query it from the API?

Comment (chenrui333) [reply Formula/d/docker.rb]: for now, I just revert the docker build change

**Review by carlocab — APPROVED** [review]

**Pull request merged**